app/editor/src/components/upload/Upload.tsx

```diff
@@ -18,6 +18,7 @@ export interface IUploadProps
   onDownload?: () => void;
   file?: IFile;
   verifyDelete?: boolean;
+  setStreamUrl?: (url: string) => void;
```

Comment:
I've been reading up on event handling in React but I'm not sure if the suggested strategies are what you expect. I want to avoid coding another solution that's not what you're expecting. Can you point me to an example in our Editor code that I can emulate?

Reply:
You're only using `setStreamUrl` to change it to `setStreamUrl?.('')';  A component should be self contained if possible and intuitive.

The basic feature you are looking for is when the `onDelete` event occurs.  So just return an `onDelete`  event call and let whatever is listening to this callback do whatever logic it wants to do.

Reply:
I believe I've addressed this.